### PR TITLE
Consumption Budget Test Fix: Create and destroy a resource group at test time

### PR DESCRIPTION
**Pull request by bgsky**
## Description

Consumption budget test uses a resource group generated and destroyed at test time. The resource group name is also randomly generated instead of hard-coded. 

## Checklist

- [X] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [X] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by bgsky** [comment]
This pull request is follow-up of [previous one](https://github.com/Azure/azure-powershell/pull/6276) to fix test issue of consumption budget. We also have an issue created [here](https://github.com/Azure/azure-powershell/issues/6439). @maddieclayton helped address the pull request and issue.

File: src/ResourceManager/Consumption/Commands.Consumption.Test/ScenarioTests/BudgetTests.ps1 [hunk 1]
```diff
@@ -135,6 +147,9 @@ function Test-BudgetAtResourceGroupLevel
 	Assert-AreEqual True $response
 
 	Assert-Throws {Get-AzureRmConsumptionBudget -Name $budgetName -ResourceGroupName $resourceGroupName}
+
+	# Remove Resource Group
+	Remove-AzureRmResourceGroup -Name $resourceGroupName -Force
```
Comment (cormacpayne) [inline src/ResourceManager/Consumption/Commands.Consumption.Test/ScenarioTests/BudgetTests.ps1]: @bgsky I would recommend wrapping the test (everything from `# Create Budget` to the command before this one) in a `try` block, and then the `finally` block would have this `Remove-AzureRmResourceGroup` call -- this ensures all resources created during this test are deleted if the test succeeds or fails

**Comment by cormacpayne** [comment]
@bgsky also, it looks like there are some additional commits in this PR, so you will need to do a [rebase](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/cleaning-up-commits.md#rebasing) to ensure the only commit(s) that appears are the ones relating to the changes in this PR

**Comment by bgsky** [comment]
@cormacpayne Thanks for your feedback. We rebased the branch, and had only one commit with a change of try-finally block. Please review and proceed on your side.

**Pull request merged**